admin/RELEASE.md

```diff
@@ -0,0 +1,89 @@
+# Release Process
+
+> Documentation guide based on the releases of `1.0.0-beta.5` on March 17, 2023.
+>
+> -kenjis
+
+## Changelog
+
+When generating the changelog each Pull Request to be included must have one of
+the following [labels](https://github.com/codeigniter4/shield/labels):
+- **bug** ... PRs that fix bugs
+- **enhancement** ... PRs to improve existing functionalities
+- **new feature** ... PRs for new features
+- **refactor** ... PRs to refactor
+- **lang** ... PRs for new/update language
+
+PRs with breaking changes must have the following additional label:
+- **breaking change** ... PRs that may break existing functionalities
+
+### Check Generated Changelog
+
+This process is checking only. Do not create a release.
+
+To auto-generate, navigate to the
+[Releases](https://github.com/codeigniter4/shield/releases) page,
+click the "Draft a new release" button.
+
+* Tag: "v1.0.0-beta.5" (Create new tag)
+* Target: develop
+
+Click the "Generate release notes" button.
+
+Check the resulting content. If there are items in the *Others* section which
+should be included in the changelog, add a label to the PR and regenerate
+the changelog.
+
+## Preparation
+
+* Clone **codeigniter4/shield** and resolve any necessary PRs
+    ```console
+    git clone git@github.com:codeigniter4/shield.git
+    ```
+* Merge any Security Advisory PRs in private forks
+
+## Process
+
+> **Note** Most changes that need noting in the User Guide and docs should have
+> been included with their PR, so this process assumes you will not be
+> generating much new content.
+
+* Create a new branch `release-1.x.x`
+* Update **src/Auth.php** with the new version number:
+  `const SHIELD_VERSION = '1.x.x';`
+* Commit the changes with "Prep for 1.x.x release" and push to origin
+* Create a new PR from `release-1.x.x` to `develop`:
+    * Title: "Prep for 1.x.x release"
+    * Description: "Updates version references for `1.x.x`." (plus checklist)
+* Let all tests run, then review and merge the PR
+* Create a new PR from `develop` to `master`:
```

Comment:
I wouldn't normally go through all this trouble for a package. Not opposed, but it makes the release harder.

Reply:
Do you mean that not creating a PR but just run `git merge`?

When releasing a package, I normally just tag in a version branch like `1.x`.